swaybar/ipc.c

```diff
@@ -606,3 +606,23 @@ bool handle_ipc_readable(struct swaybar *bar) {
 	free_ipc_response(resp);
 	return bar_is_dirty;
 }
+
+json_object *ipc_get_keyboard_modifiers(struct status_line *status) {
+	uint32_t len = 0;
+	char *buf = ipc_single_command(status->bar->ipc_socketfd,
+			IPC_GET_INPUTS, NULL, &len);
```

Comment:
We don't need to use the Sway IPC to get this information. Instead, we can just use Wayland. We'd need to add a `wl_keyboard` listener in `swaybar/input.c`.

Reply:
Will it work, since swaybar surface is not in focus? I tried attaching `wl_keyboard_listener` with a bunch of logging inside callbacks, but they aren't fired when I press keys.

Reply:
Here's my current swaybar/input.c: https://gist.github.com/kraftwerk28/aedafb935d80638dc3f08e51a71a187c

https://user-images.githubusercontent.com/31807671/145615521-16854db0-6363-4d74-b152-23e988561b8a.mp4